bag_replicator/src/main/scala/uk/ac/wellcome/platform/archive/bagreplicator/Main.scala

```diff
@@ -70,7 +72,16 @@ object Main extends WellcomeTypesafeApp {
     val lockingService =
       new DynamoLockingService[IngestStepResult[BagReplicationSummary[_]], Try]()
 
-    val replicator = new S3Replicator()
+    val replicator = config.requireString("bag-replicator.provider") match {
+      case "amazon-s3"          => new S3Replicator()
+      case "azure-blob-storage" =>
```

Comment:
You could use: https://github.com/wellcomecollection/scala-libs/blob/master/storage_typesafe/src/main/scala/uk/ac/wellcome/storage/typesafe/AzureBlobServiceClientBuilder.scala#L7

Reply:
Ah, I’ve not committed the change, but connectionString turns out to be the wrong thing for SAS tokens – this should be `endpoint()`.

But good shout, we should definitely get this into the libs.